HDSAviz/data_processing.py

```diff
@@ -0,0 +1,218 @@
+    drop=False.
+
+    Parameters:
+    ----------
+    numrows : the number of rows of the input_parameters file to read
+              (default is to read all rows).
+    drop    : a list of strings for which parameters you do not want to
+              include in the returned dataframe.  If you want all params
+              then pass drop=False.
+
+    Returns:
+    -------
+    A pandas dataframe
+
+    """
+    return read_file('../../HDSAviz_data/input_parameters.csv',
+                     numrows=numrows, drop=drop)
+
+
+def get_results(numrows=None, drop=['light_aromatic_C-C',
+                                    'light_aromatic_methoxyl']):
+    """
+    NOTE: This function is specific to our lignin modeling dataset
+
+    Returns a pandas dataframe with the results of running all of the
+    simulations for the parameters sets in `input_parameters.csv`. This
+    function drops two unused functional groups from the results file.
+
+    Parameters:
+    ----------
+    numrows : the number of rows of the input_parameters file to read
+              (default is to read all rows).
+    drop    : a list of strings for which output measures to drop from
+              the returned dataframe.  If you want all outputs use
+              drop=False.
+    """
+    return read_file('../../HDSAviz_data/results.csv',
+                     numrows=numrows, drop=drop)
+
+
+def get_sa(path='../../HDSAviz_data/'):
+    """
+    This function reads and processes all the sensitivity analysis results
+    in a specified folder, and returns a dictionary with the corresponding
+    dataframes.  Sensitivity analysis results should be in the default SALib
+    format and must start with the letter 'a'.
+
+    NOTE: there are two lines of code at the beginning of this function
+    (the filenames.remove lines) that are specific to our lignin modeling
+    dataset.  Future users will want to remove or modify these lines to use
+    with other datasets.
+
+    Parameters:
+    -----------
+    path : String containing the relative or absolute path of the directory
+           where analysis_*.txt files are stored.  There cannot be any
+           files or folders within this directory that start with 'a'
+           except those generated by the SALib sensitivity analysis.
+
+    Returns:
+    --------
+    sens : Dictionary where keys are the names of the various output measures
+           (one output measure per analysis file in the folder specified by
+           path).  Dictionary values are a list of pandas dataframes.  If
+           only first and total order indices were calculated in the
+           sensitivity analysis then this list will contain just one pandas
+           dataframe (sens[key][0]) and the second item in the list will be
+           False.  If second order indices are present they are in a second
+           dataframe (sens[key][1]).
+    """
+    filenames = [filename for filename in os.listdir(
```

Comment:
I think you can be more specific on startswith('analysis_') but this seems good with current data set.